### PR TITLE
Issue #92: Player.cs has extraneous code

### PR DESCRIPTION
**Pull request by maratmuginov**
Major refactoring to reduce redundancy, improve performance with readonly struct's and introduce short local functions in compliance with clean code principles.

### CODE REVIEW

File: Core/GameObjects/Map/EmberGrid.cs [hunk 1]
```diff
@@ -226,23 +226,44 @@ public void UpdateFieldOfView()
...
+
+            var cells = GetCellsInFov(entity);
+
+            entity.FieldOfViewRadius = originalFov;
+            EntityManager.RecalculatFieldOfView(entity, false);
+
             return cells;
         }
 
+        public IEnumerable<EmberCell> GetExploredCells(IEntity entity)
```
Comment (Venom0us) [inline Core/GameObjects/Map/EmberGrid.cs:257]: Shouldn't this also be called GetExploredCellsInFov as that's what it seems to do?

File: Core/UserInterface/Windows/FovWindow.cs [hunk 2]
```diff
@@ -92,52 +111,78 @@ private Color GetColorByString(string value)
...
+        {
+            _textConsole.Cursor.Print(new ColoredString('[' + charObj.Glyph + "]:", charObj.GlyphColor, Color.Transparent));
+            _textConsole.Cursor.Print(' ' + charObj.Name);
+            _textConsole.Cursor.CarriageReturn();
+            _textConsole.Cursor.LineFeed();
+        }
+
+        public void Update(IEntity entity)
+        {
+            int prevFov = entity.FieldOfViewRadius;
```
Comment (Venom0us) [inline Core/UserInterface/Windows/FovWindow.cs:143]: prevFov no longer needed

File: Core/GameObjects/Entities/Player.cs [hunk 3]
```diff
@@ -53,48 +48,15 @@ public override void OnMove(object sender, EntityMovedEventArgs args)
...
-            FieldOfViewRadius = Constants.Player.FieldOfViewRadius + 3;
-            EntityManager.RecalculatFieldOfView(this, false);
+            _fovObjectsWindow ??= UserInterfaceManager.Get<FovWindow>();
+            _fovObjectsWindow.Update(this);
+        }
 
-            // Get first the further bright cells
-            var farBrightCells = GridManager.Grid.GetCellsInFov(this)
-                .Where(a => a.CellProperties.IsExplored)
+        private IEnumerable<char> GetBrightCellsInFov(int fovRadius)
```
Comment (Venom0us) [inline Core/GameObjects/Entities/Player.cs:55]: I think this method is no longer of use in the Player class?

File: Core/GameObjects/Entities/Player.cs [hunk 4]
```diff
@@ -53,48 +48,15 @@ public override void OnMove(object sender, EntityMovedEventArgs args)
 
         private void UpdateFovWindow()
```
Comment (Venom0us) [inline Core/GameObjects/Entities/Player.cs:49]: Can we get rid of this method, initialize FovWindow in the constructor of player entity and call 
_fovObjectsWindow.Update(this); in OnMove override instead?

File: Core/GameObjects/Map/EmberGrid.cs (hunk 1 above)
Comment (maratmuginov) [reply Core/GameObjects/Map/EmberGrid.cs:257]: I thought it should be named GetExploredCells() since it doesn't take in fovRadius.

Comment (Venom0us) [reply Core/GameObjects/Map/EmberGrid.cs:257]: It's still getting only the cells from the Fov (`GetCellsInFov(entity)`), so the naming is wrong.

File: Core/UserInterface/Windows/FovWindow.cs [hunk 5]
```diff
@@ -1,4 +1,5 @@
-﻿using Emberpoint.Core.Extensions;
+﻿using System;
```
Comment (Venom0us) [inline Core/UserInterface/Windows/FovWindow.cs:1]: This one is redundant

File: Core/UserInterface/Windows/FovWindow.cs [hunk 6]
```diff
@@ -92,52 +90,70 @@ private Color GetColorByString(string value)
...
+        {
+            _textConsole.Cursor.Print(new ColoredString('[' + charObj.Glyph + "]:", charObj.GlyphColor, Color.Transparent));
+            _textConsole.Cursor.Print(' ' + charObj.Name);
+            _textConsole.Cursor.CarriageReturn();
+            _textConsole.Cursor.LineFeed();
+        }
+
+        public void Update(IEntity entity)
+        {
+            var farBrightCells = GetBrightCellsInFov(entity, Constants.Player.FieldOfViewRadius + 3).Distinct();
```
Comment (Venom0us) [inline Core/UserInterface/Windows/FovWindow.cs:122]: distinct is redundant

File: Core/GameObjects/Map/EmberGrid.cs [hunk 7]
```diff
@@ -226,23 +226,44 @@ public void UpdateFieldOfView()
...
             for (int x = 0; x < GridSizeX; x++)
             {
                 for (int y = 0; y < GridSizeY; y++)
                 {
                     var cell = GetNonClonedCell(x, y);
                     if (entity.FieldOfView.BooleanFOV[x, y])
-                    {
-                        cells.Add(cell);
-                    }
+                        yield return cell.Clone();
```
Comment (Venom0us) [inline Core/GameObjects/Map/EmberGrid.cs:237]: We don't want delayed execution, best to return a new list here instead.
IEnumerable as return type can stay that's fine.

**Comment by Venom0us** [comment]
Looks good, thanks for the contribution!

**Pull request merged**